sim/mage/talents.go

```diff
@@ -97,9 +98,11 @@ func (mage *Mage) applyArcaneConcentration() {
 					return
 				}
 
-				mage.AddAura(sim, ccAura)
-				// Also has special interaction with AM, gets the benefit of CC crit bonus on its own cast.
-				cast.BonusCritRating += bonusCrit
+				if !mage.HasAura(ClearcastingAuraID) {
+					// Also has special interaction with AM, gets the benefit of CC crit bonus on its own cast.
+					cast.BonusCritRating += bonusCrit
+				}
+				mage.ReplaceAura(sim, ccAura)
```

Comment:
I haven't looked much deeper, but it seems like clearcasting procced twice on a spell cast, which sounds wrong.